src/hooks/useRedeemControl.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ChainId,
+  ParsedVaa,
+  parseTransferPayload,
+  parseVaa,
+  hexToUint8Array,
+  tryHexToNativeString,
+} from "@certusone/wormhole-sdk";
+import { Rule, WarningMessage } from "./useTransferControl";
+import { useEffect, useMemo, useState } from "react";
+
+export function useRedeemControl(
+  rules: Rule[],
+  sourceChain: ChainId,
+  targetChain: ChainId,
+  rawVaa: string = ""
+) {
+  const [ids, setIds] = useState<string[]>([]);
+  const [isTransferDisabled, setIsTransferDisabled] = useState<boolean>(false);
+  const [warnings, setWarnings] = useState<WarningMessage[]>([]);
+  const vaa: ParsedVaa | null = useMemo(() => {
+    try {
+      return parseVaa(hexToUint8Array(rawVaa));
+    } catch (e) {
+      console.error(e);
+    }
+    return null;
+  }, [rawVaa]);
+  const payload = useMemo(() => {
+    try {
+      if (vaa) {
+        return parseTransferPayload(Buffer.from(new Uint8Array(vaa?.payload)));
+      }
+    } catch (e) {
+      console.error(e);
+    }
+    return null;
+  }, [vaa]);
+  const asset = useMemo(
```

Comment:
Do we need the logic to parse the asset? Could it be already available on the store? I see an [entry](https://github.com/XLabs/portal-bridge-ui/blob/main/src/store/transferSlice.ts#L45) for it in the state, unless it's not what we need. If such data is already available we might be able to create a single hook `useOperationControl` since the rest of the code seems to be the same.

Mind you, I'm commenting without the full context. You might have already ruled this option for some reason.

Reply:
I've tried to reuse it, but, the state does not have all the info, and can not store a Buffer or ByteArray like structure, so, I've opted in to replicate a little bit here